src/project-css/app/Http/Controllers/DataViewController.php

```diff
@@ -35,14 +35,45 @@ public function index(Request $request, $curTable){
       return redirect()->route('home')->withErrors(['Table is disabled']);
     }
 
+    /**
+    * set passed values to variables
+    */
+    $search = $request->input('search');
+    $tblCol = $request->input('tblCol');
+    $id = $request->input('id');
+
+    if ((strlen($search) != '0') && (strlen($tblCol) != '0')) {
+      $numOfRcrds = DB::table($curTable->tblNme)
+                      ->where($tblCol, 'LIKE', $search)
+                      ->count();
+    }
+    else {
+      $numOfRcrds = DB::table($curTable->tblNme)->count();
+    }
+
     // check for the number of records
-    $numOfRcrds = DB::table($curTable->tblNme)->count();
     if ($numOfRcrds == 0){
       return redirect()->route('home')->withErrors(['Table does not have any records.']);
     }
 
     // Get the records of the table using the name of the table we are currently using
-    $rcrds = DB::table($curTable->tblNme)->paginate(25);
+    if ((strlen($search) != '0') && (strlen($tblCol) != '0')) {
```

Comment:
Why is this duplicated on line 45?  Needs refactored if there are going to be multiple instances calling a like method.